csharp/Adapter/Microsoft.Spark.CSharp/Sql/DataFrame.cs

```diff
@@ -26,14 +24,14 @@ public class DataFrame
         private readonly SparkContext sparkContext;
         [NonSerialized]
         private StructType schema;
-        private RowSchema rowSchema;
         [NonSerialized]
         private RDD<Row> rdd;
-
+        [NonSerialized]
+        private IRDDProxy rddProxy;
         [NonSerialized]
         private bool? isLocal;
         [NonSerialized]
-        private Random random = new Random();
+        private readonly Random random = new Random();
```

Comment:
Not directly related to to this change, but it is very important to remember that the `Random` class is NOT thread-safe.
https://msdn.microsoft.com/en-us/library/system.random(v=vs.110).aspx#ThreadSafety

To make things even worse, the side effect of multi-threaded usage of an instance of `Random` class is a massively degenerate failure mode -- it starts to always return `0` on each call ! -- ie, about as skewed and far away from "random" as you could possibly get!

We got hit by that really nasty "feature" of `Random` in Orleans project, and took a long time to work out what was going wrong. 
I fear that client code passing around `DataFrame` object references could easily trigger this same "feature". :(

In Orleans we ended up creating a `SafeRandom` wrapper as a drop-in replacement, which i would be happy to send across as a separate PR if you want?
https://github.com/dotnet/orleans/blob/a64fa8953e764ac391ebd549d395ac66374b545f/src/Orleans/Utils/SafeRandom.cs